posts/deploy-coder-on-fly-io/index.md

```diff
@@ -0,0 +1,106 @@
+# Deploy Coder on Fly.io
+
+Fly.io is a platform for deploying applications to the edge. It's a great fit for Coder because it's easy to deploy and scale, and it's easy to manage. In this guide, we'll walk through deploying Coder on Fly.io.
+
+## Prerequisites
+
+- A Fly.io [account](https://fly.io/signup).
+- [Fly CLI](https://fly.io/docs/getting-started/installing-flyctl/) installed on your machine.
+
+## Deploy Coder
+
+1. Log in to Fly.io with the CLI:
+
+```bash
+flyctl auth login
+```
+
+2. Create a new fly postgres database:
+
+```bash
+flyctl postgres create DATABASE-NAME
+```
+
+3. Create a new fly app:
+
+```bash
+flyctl apps create APP-NAME
+```
+
+4. Connect to the database with the coder fly app:
+
+```bash
+flyctl postgres connect DATABASE-NAME --app APP-NAME
+```
+
+This will print out a connection string. Copy the connection string and save it to be used in the next step.
+
+5. Edit the `fly.toml` file and update as per the example below:
+
+```toml
+app = "APP-NAME"
+kill_signal = "SIGINT"
+kill_timeout = 5
+primary_region = "ams"  # See a list of regions here: https://fly.io/docs/reference/regions/
+
+[experimental]
+  auto_rollback = true
+  private_network = true  # This is needed for Coder to connect to the database
+
+[build]
+   image = "ghcr.io/coder/coder:latest"
+
+[env]
+  CODER_ACCESS_URL = "https://APP-NAME.fly.dev" # make sure to replace APP-NAME with your app name
```

Comment:
Fly will inject `FLY_APP_NAME` here, so that should save a step. https://fly.io/docs/reference/runtime-environment/

I think you can do this, but you might want to confirm:

```suggestion
  CODER_ACCESS_URL = "https://$FLY_APP_NAME.fly.dev" # make sure to replace APP-NAME with your app name
```

Reply:
I tested and it seems like these are internel varibales and are not passed to Coder. A user has to set it manully.

Reply:
It seems like we can not use secrets in fly.toml file. Not working for my testing.